pkg/sql/drop_index.go

```diff
@@ -135,6 +137,16 @@ func (p *planner) dropIndexByName(
 	if dropped {
 		return nil
 	}
+	if behavior == tree.DropRestrict {
+		name, constraint, err := findIndexUniqueConstraint(ctx, p.txn, tableDesc, &idx)
```

Comment:
TBH I'm not too happy with this, because it's getting all constraints for the table for each index that is being dropped. It may be possible to bring this above this function to only fetch constraints for any tables involved once.